src/Graphing.hs

```diff
@@ -92,13 +92,17 @@ edge parent child gr = gr { graphingAdjacent = adjacent' }
 -- - A way to convert a dependency @toDependency@
 --
 -- __Unfold does not work for recursive inputs__
-unfold :: Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
-unfold seed getDeps toDependency = foldr addNode empty seed
+unfold :: forall dep res. Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
```

Comment:
Why do we need to add `forall dep res` here?

Reply:
By default, this `forall` is automatically generated whenever you use a type variable, e.g., if I define
```hs
const :: a -> b -> a
```
the `forall` is implicit:
```hs
const :: forall a b. a -> b -> a
```

with `-XScopedTypeVariables`, the use of the explicit forall here means I can use `dep` in the type signature of `edgesFrom` in the `where` clause below.

---

in rust, `forall` is analogous to (pseudocode?):
`fn const<A,B>(a: A, b: B) -> A`